### PR TITLE
When You Delete An Audio Message While It's Playing it will Still Plays

### PR DESCRIPTION
**Pull request by gitstart**
### First time contributor checklist

- [x] I have read [how to contribute](https://github.com/signalapp/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/signalapp/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Pixel 6, Android 13
 * Virtual device Pixel 4a, Android 12
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### Description

fixes https://github.com/signalapp/Signal-Android/issues/10810

Bug -> https://www.loom.com/share/28c2baea553b461098886f1444f6a744
Fix-> https://www.loom.com/share/56675f70de564dc6a7bbf110c0ac37e4

when a voice note is playing and user attempt to delete the voice note, the voice not continues playing in the background. 

After this changes when the voice is deleted, it no longer plays in the background

### CODE REVIEW

File: app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java [hunk 1]
```diff
@@ -1096,11 +1099,24 @@ private static boolean isNoteToSelfDelete(Set<MessageRecord> messageRecords) {
...
+    }
+  }
+
   private void handleDeleteForEveryone(Set<MessageRecord> messageRecords) {
     Runnable deleteForEveryone = () -> {
       SignalExecutors.BOUNDED.execute(() -> {
         for (MessageRecord message : messageRecords) {
           MessageSender.sendRemoteDelete(message.getId(), message.isMms());
+          if (message.isMms()){
+            handleStopAudioMessageAfterDeletion((MediaMmsMessageRecord)message);
```
Comment (alex-signal) [inline app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java:1118]: `(MediaMmsMessageRecord)message`

Please add a space between cast and object

File: app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java [hunk 2]
```diff
@@ -1065,6 +1067,7 @@ protected Void doInBackground(Void... voids) {
 
             if (messageRecord.isMms()) {
               threadDeleted = SignalDatabase.mms().deleteMessage(messageRecord.getId());
+              handleStopAudioMessageAfterDeletion((MediaMmsMessageRecord)messageRecord);
```
Comment (alex-signal) [inline app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java:1070]: `(MediaMmsMessageRecord)message`

Please add a space between cast and object

File: app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java [hunk 3]
```diff
@@ -1046,6 +1047,7 @@ private void handleDeleteMessages(final Set<MultiselectPart> multiselectParts) {
   }
 
   private AlertDialog.Builder buildRemoteDeleteConfirmationDialog(Set<MessageRecord> messageRecords) {
+    Context             context       = requireActivity();
```
Comment (alex-signal) [inline app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java:1050]: Where is this used?

File: app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java [hunk 4]
```diff
@@ -1096,11 +1099,24 @@ private static boolean isNoteToSelfDelete(Set<MessageRecord> messageRecords) {
     return messageRecords.stream().allMatch(messageRecord -> messageRecord.isOutgoing() && messageRecord.getRecipient().isSelf());
   }
 
+  private void handleStopAudioMessageAfterDeletion(@NonNull MediaMmsMessageRecord message){
```
Comment (alex-signal) [inline app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java:1102]: Something about this fix makes me wonder if we shouldn't be handling message deletion in the `VoiceNotePlaybackService`.

We could add a database notifier in there that lasts for the lifetime of the service, and if we see that the current playing item is deleted, we can stop playback at that point. That would cover things like expiring messages and would also cover remote deletion by another user.

**Review by alex-signal — CHANGES_REQUESTED** [review]
A few nitpicks, but I'm going to share this with the team and see what they think of the approach.

File: app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java (hunk 4 above)
Comment (gitstart) [reply app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java:1102]: Handling message deletion seems to be outside the scope of this PR, what do you think?

Comment (alex-signal) [reply app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationFragment.java:1102]: This PR is all about message deletion, and this feels like a partial fix to me. Having the service handle the deletion would make more sense, you wouldn't need any of this code here, you'd just need to handle it properly in the service itself.

**Pull request closed without merging**